packages/tpetra/core/example/Finite-Element-Assembly/fem_assembly_type3_staticprofile.cpp

```diff
@@ -0,0 +1,322 @@
+    }
+    for(size_t element_node_idx=0; element_node_idx<ghost_element_to_node_ids.extent(1); element_node_idx++)
+    {
+      if(mesh.nodeIsOwned(global_ids_in_row[element_node_idx]))
+      {
+       crs_graph->insertGlobalIndices(global_ids_in_row[element_node_idx], global_ids_in_row());
+      }
+    }
+  }
+
+  timerElementLoopGraph = Teuchos::null;
+
+  // 'finalize' the crs_graph by calling fillComplete().
+  {
+    RCP<TimeMonitor> timerFillCompleteGraph = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("2) FillComplete (Graph)")));
+    crs_graph->fillComplete();
+  }
+
+  // Print out the crs_graph in detail...
+  #if PRINT_VERBOSE
+  crs_graph->describe(*out, Teuchos::VERB_EXTREME);
+  #endif
+
+  // Matrix Fill
+  // -------------------
+  // In this example, we're using a simple stencil of values for the stiffness matrix:
+  //
+  //    +-----+-----+-----+-----+
+  //    |  2  | -1  |     | -1  |
+  //    +-----+-----+-----+-----+
+  //    | -1  |  2  | -1  |     |
+  //    +-----+-----+-----+-----+
+  //    |     | -1  |  2  | -1  |
+  //    +-----+-----+-----+-----+
+  //    | -1  |     | -1  |  2  |
+  //    +-----+-----+-----+-----+
+  //
+  // For matrix fill, we create the crs_matrix object and will fill it 
+  // in the same manner as we filled in the graph but in this case, nodes 
+  // associated with each element will receive contributions according to 
+  // the row in this stencil.
+  //
+  // In this example, the calls to sumIntoGlobalValues() on 1 core will look like:
+  //   Element 0
+  // - sumIntoGlobalValues( 0,  [  0  1  5  4  ],  [  2  -1  0  -1  ])
+  // - sumIntoGlobalValues( 1,  [  0  1  5  4  ],  [  -1  2  -1  0  ])
+  // - sumIntoGlobalValues( 5,  [  0  1  5  4  ],  [  0  -1  2  -1  ])
+  // - sumIntoGlobalValues( 4,  [  0  1  5  4  ],  [  -1  0  -1  2  ])
+  // Element 1
+  // - sumIntoGlobalValues( 1,  [  1  2  6  5  ],  [  2  -1  0  -1  ])
+  // - sumIntoGlobalValues( 2,  [  1  2  6  5  ],  [  -1  2  -1  0  ])
+  // - sumIntoGlobalValues( 6,  [  1  2  6  5  ],  [  0  -1  2  -1  ])
+  // - sumIntoGlobalValues( 5,  [  1  2  6  5  ],  [  -1  0  -1  2  ])
+  // Element 2
+  // - sumIntoGlobalValues( 2,  [  2  3  7  6  ],  [  2  -1  0  -1  ])
+  // - sumIntoGlobalValues( 3,  [  2  3  7  6  ],  [  -1  2  -1  0  ])
+  // - sumIntoGlobalValues( 7,  [  2  3  7  6  ],  [  0  -1  2  -1  ])
+  // - sumIntoGlobalValues( 6,  [  2  3  7  6  ],  [  -1  0  -1  2  ])
+  //
+  // Similarly to the Graph construction above, we loop over both local and global
+  // elements and insert rows for only the locally owned rows.
+  //
+  RCP<TimeMonitor> timerElementLoopMatrix = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("3) ElementLoop  (Matrix)")));
+
+  RCP<MatrixType> crs_matrix = rcp(new MatrixType(crs_graph));
+
+  scalar_2d_array_type element_matrix;
+  Kokkos::resize(element_matrix, 4, 4);
+
+  Teuchos::Array<GlobalOrdinal> column_global_ids(4);     // global column ids list
```

Comment:
See notes above on preferring `std::vector` to `Teuchos::Array`.  `CrsMatrix` has overloads that would work for this.

Reply:
It's OK for now.  We can go back and adjust this later.

It's not entirely clear what this interface will look like in the future.  The C++ `span` and related proposals look very promising, and I'm hoping at some point that they will let us replace `Teuchos::ArrayView`.

Reply:
ok.  I'll hold off on making changes to this for now.